python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/sfn.py

```diff
@@ -0,0 +1,160 @@
+    from mypy_boto3_stepfunctions import SFNClient
+    from mypy_boto3_stepfunctions.type_defs import (
+        DescribeExecutionOutputTypeDef,
+        StartExecutionInputRequestTypeDef,
+    )
+
+
+@experimental
+class PipesSFNClient(PipesClient, TreatAsResourceParam):
+    """A pipes client for invoking AWS Step Functions.
+
+    Args:
+        context_injector (Optional[PipesContextInjector]): A context injector to use to inject
+            context into the Step Function execution.
+        message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
+            from the Step Function execution. Defaults to :py:class:`PipesCloudWatchMessageReader`.
+        client (Optional[boto3.client]): The boto Step Functions client used to start the execution.
+        forward_termination (bool): Whether to cancel the Step Function execution when the Dagster process receives a termination signal.
+    """
+
+    def __init__(
+        self,
+        context_injector: PipesContextInjector,
+        message_reader: Optional[PipesMessageReader] = None,
+        client: Optional["SFNClient"] = None,
+        forward_termination: bool = True,
+    ):
+        self._client: SFNClient = client or boto3.client("stepfunctions")
+        self._context_injector = context_injector
+        self._message_reader = message_reader or PipesCloudWatchMessageReader()
+        self.forward_termination = check.bool_param(forward_termination, "forward_termination")
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return True
+
+    @public
+    def run(
+        self,
+        *,
+        context: Union[OpExecutionContext, AssetExecutionContext],
+        start_execution_input: "StartExecutionInputRequestTypeDef",
+        extras: Optional[Dict[str, Any]] = None,
+    ) -> PipesClientCompletedInvocation:
+        """Start a Step Function execution, enriched with the pipes protocol.
+
+        See also: `AWS API Documentation <https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartExecution.html>`_
+
+        Args:
+            context (Union[OpExecutionContext, AssetExecutionContext]): The context of the currently executing Dagster op or asset.
+            start_execution_input (Dict): Parameters for the ``start_execution`` boto3 Step Functions client call.
+            extras (Optional[Dict[str, Any]]): Additional Dagster metadata to pass to the Step Function execution.
+
+        Returns:
+            PipesClientCompletedInvocation: Wrapper containing results reported by the external
+            process.
+        """
+        params = start_execution_input
+
+        state_machine_arn = cast(str, params["stateMachineArn"])
+
+        with open_pipes_session(
+            context=context,
+            message_reader=self._message_reader,
+            context_injector=self._context_injector,
+            extras=extras,
+        ) as session:
+            _ = session.get_bootstrap_cli_arguments()
```

Comment:
We probably have to use some mapping to pass our bootstrap info to the external process. In Lambda we are doing it via Lambda args.